src/test/java/GoblintMessagesTest.java

```diff
@@ -163,5 +172,61 @@ public void testConvertMessagesFromJson() throws IOException {
     }
 
 
+
+
+    @Test
+    public void testConvertFunctionsFromJson() throws IOException {
+        List<GoblintFunctionsResult> goblintFunctionsResults = readGoblintResponseJsonFunc();
+
+        when(goblintService.functions()).thenReturn(CompletableFuture.completedFuture(goblintFunctionsResults));
+        when(gobPieConfiguration.showCfg()).thenReturn(false);
+
+        goblintAnalysis.analyze(files, analysisConsumer, true);
+
```

Comment:
`goblintService.messages()` should be mocked to return a `CompletableFuture` with an empty list here, so that the response would only contain the converted functions.